### DISCUSSION TITLE
using SWR to implement conversation between 2 users

### DISCUSSION THREAD

**Discussion by oran1248** [vercel/swr#458] [General]
I want to use SWR to show and refresh every X seconds a conversation component between 2 users.
But i don't want that SWR will fetch every time all the messages, but always fetch for new ones.
How can I do it? because every time new data arrives it will override the old one.
Any ideas?
BTW, im using redux if it helps somehow

Another question, is it make sense to update redux state when fetching with swr? it will not cause an infinite rendering loop?

**Comment by sergiodxa** [comment]
You can use mutate to update the cached value, there is a PR (https://github.com/vercel/swr/pull/457) to add native subscription support which will solve your issue.

**Reply by oran1248** [reply]
@sergiodxa Yes but I don't use realtime db for the messages. I use mongodb. Is it possible to do it with mongo?  So if I understand correctly, for now there is no way to fetch for only "new data".

**Reply by sergiodxa** [reply]
You don't need a real-time DB to build a WebSockets server and notify users when a new message comes.

SWR doesn't support fetching specific keys, even with subscription support it will still not support it, I would recommend you to split more your data fetching, fetch the list of IDs and then fetch each message in their own component using useSWR, this way your list will be smaller because it's a list of ids and you can individually update each message without refetching the whole list.

**Reply by oran1248** [reply]
Sounds (to me) unefficient to use swr for each message. Messages content does not change at all (there is no need to revalidate). Also i need to support infinitescroll so swr cant help me to fetch more and more old messages and also update when new one arrives. Im guessing this scenario of chat between 2 users is too complicated for swr. Am i missing something?

**Reply by CodeLuca** [reply]
curious what you ended up doing @oran1248

**Reply by oran1248** [reply]
@CodeLuca Didn't use SWR for this :)

**Reply by rjjfajardo** [reply]
@oran1248 this might help you if you are using mongodb  https://docs.mongodb.com/manual/changeStreams/
